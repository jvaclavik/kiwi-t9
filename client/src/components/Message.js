// @flow
import React from "react"

// components
import { ScrollView, Text } from "react-native"

// utils
import renderStyle from "../utils/renderStyle"

// theme
import { Metrics, Fonts } from "../themes"

const styles = {
  container: {
    flex: 1,
    paddingHorizontal: Metrics.spacings.section,
  },
  text: {
    fontSize: Fonts.sizes.subtitle,
  },
}

type Props = {|
  +message: string,
|}

export default class Message extends React.PureComponent<Props> {
  render() {
    const { message } = this.props
    return (
      <ScrollView style={renderStyle(styles.container)}>
        <Text style={renderStyle(styles.text)}>{message}</Text>
      </ScrollView>
    )
  }
}
