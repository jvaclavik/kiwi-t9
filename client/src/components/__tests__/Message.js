// @flow
import React from "react"
import renderer from "react-test-renderer"

// components
import Message from "../Message"

test("renders correctly with required props only", () => {
  const tree = renderer.create(<Message message="ahojky" />).toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders correctly with optional props", () => {
  const tree = renderer
    .create(<Message message="ahojky" style={{ backgroundColor: "white" }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
