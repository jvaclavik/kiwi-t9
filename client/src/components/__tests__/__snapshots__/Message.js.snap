// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with optional props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
    }
  }
/>
`;

exports[`renders correctly with required props only 1`] = `
<View
  style={Object {}}
/>
`;
